microprofile.jdt/com.redhat.microprofile.jdt.core/src/main/java/com/redhat/microprofile/jdt/core/java/IJavaDiagnosticsParticipant.java

```diff
@@ -0,0 +1,55 @@
+/*******************************************************************************
+* Copyright (c) 2020 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.core.java;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.JavaModelException;
+
+/**
+ * Java diagnostics participants API.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public interface IJavaDiagnosticsParticipant {
+
+	/**
+	 * Returns true if the collection of diagnostics must be done for the given
+	 * context and false otherwise.
+	 * 
+	 * <p>
+	 * Collection is done by default. Participants can override this to check if
+	 * some classes are on the classpath before deciding to process the collection.
+	 * </p>
+	 * 
+	 * @param the     java diagnostics context
+	 * @param monitor the progress monitor
+	 * @return true if the collection of diagnostics must be done for the given

```

Comment:
true if diagnostics must be collected for the ...

Reply:
fixed.